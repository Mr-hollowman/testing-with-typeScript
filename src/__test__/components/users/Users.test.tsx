import { render, screen } from "@testing-library/react";
import Users from "../../../components/users/Users";
import { server } from "../../../mocks/server";
import { rest } from 'msw'

describe("Users component", ()=>{
    test("renders correctly",()=>{
        render(<Users />)
        const headingElement = screen.getByRole("heading",{level:1})
        expect(headingElement).toBeInTheDocument()
    })

    test("renders correctly with list of users", async ()=>{
        render(<Users />)
        const listElements = await screen.findAllByRole("listitem")
        expect(listElements).toHaveLength(3)
    })

    test("render error", async ()=>{
        server.use(
            rest.get("https://jsonplaceholder.typicode.com/users", (req, res, ctx)=>{
                return res(ctx.status(500))
            })
        )

        render(<Users />)
        const error = await screen.findByText("error fetching users")
        expect(error).toBeInTheDocument()
    })

    test("snapshot testing", ()=>{
        const {container} = render(<Users />)
        expect(container).toMatchSnapshot()
    })
})