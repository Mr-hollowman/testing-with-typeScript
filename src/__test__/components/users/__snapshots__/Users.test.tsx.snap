// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component snapshot testing 1`] = `
<div>
  <div>
    <h1>
      Users
    </h1>
    <ul />
  </div>
</div>
`;
